connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigStorage.java

```diff
@@ -396,6 +398,7 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
                 synchronized (lock) {
                     if (value.value() == null) {
                         // Connector deletion will be written as a null value
+                        log.info("Removed connector " + connectorName + " due to null configuration. This is usually intentional and does not indicate an issue.");
```

Comment:
Deletions will eventually be compacted away, so I figured it should not be too painful. 
IMO a connector completely going away is a big event that someone watching logs will want to know about - which is why I chose INFO. Maybe I am underestimating the amount of log crap that will show up on restart - I imagined that each connector will have 5-10 connectors, so we are talking about ~ 10 messages at startup. Do you foresee different numbers? Or you think the 10 extra messages are already too much?